docs/hardware_keyboard_guidelines.md

```diff
@@ -88,12 +88,14 @@ The `config.h` files can also be placed in sub-folders, and the order in which t
       * `keyboards/top_folder/sub_1/sub_2/sub_3/config.h`
         * `keyboards/top_folder/sub_1/sub_2/sub_3/sub_4/config.h`
           * `users/a_user_folder/config.h`
-          * `keyboards/top_folder/keymaps/a_keymap/config.h`
-        * `keyboards/top_folder/sub_1/sub_2/sub_3/sub_4/post_config.h`
-      * `keyboards/top_folder/sub_1/sub_2/sub_3/post_config.h`
-    * `keyboards/top_folder/sub_1/sub_2/post_config.h`
-  * `keyboards/top_folder/sub_1/post_config.h`
-* `keyboards/top_folder/post_config.h`
+            * `keyboards/top_folder/keymaps/a_keymap/config.h`
+          * `keyboards/top_folder/sub_1/sub_2/sub_3/sub_4/post_config.h`
+        * `keyboards/top_folder/sub_1/sub_2/sub_3/post_config.h`
+      * `keyboards/top_folder/sub_1/sub_2/post_config.h`
+    * `keyboards/top_folder/sub_1/post_config.h`
+  * `keyboards/top_folder/post_config.h`
+* `users/a_user_folder/post_config.h`
+
```

Comment:
There are some undocumented post_config.h added in common_features.mk.
I propose these clarifications.

```suggestion
* System side post_config.h.

```

And I propose the following changes to improve the readability of build_keyboard.mk and common_features.mk.
```diff
diff --git a/build_keyboard.mk b/build_keyboard.mk
index a235d9b807..485de45228 100644
--- a/build_keyboard.mk
+++ b/build_keyboard.mk
@@ -407,6 +407,7 @@ VPATH += $(USER_PATH)
 VPATH += $(KEYBOARD_PATHS)
 VPATH += $(COMMON_VPATH)
 
+COMMON_FEATURES_POST_CONFIG_H :=
 include common_features.mk
 include $(TMK_PATH)/protocol.mk
 include $(TMK_PATH)/common.mk
@@ -438,7 +439,7 @@ PROJECT_DEFS := $(OPT_DEFS)
 PROJECT_INC := $(VPATH) $(EXTRAINCDIRS) $(KEYBOARD_PATHS)
 PROJECT_CONFIG := $(CONFIG_H)
 
-CONFIG_H += $(POST_CONFIG_H)
+CONFIG_H += $(POST_CONFIG_H) $(COMMON_FEATURES_POST_CONFIG_H)
 ALL_CONFIGS := $(PROJECT_CONFIG) $(CONFIG_H)
 
 OUTPUTS := $(KEYMAP_OUTPUT) $(KEYBOARD_OUTPUT)
diff --git a/common_features.mk b/common_features.mk
index 6a9754e58e..029af48b96 100644
--- a/common_features.mk
+++ b/common_features.mk
@@ -191,7 +191,7 @@ ifeq ($(strip $(RGBLIGHT_ENABLE)), yes)
         $(error RGBLIGHT_DRIVER="$(RGBLIGHT_DRIVER)" is not a valid RGB type)
     else
         COMMON_VPATH += $(QUANTUM_DIR)/rgblight
-        POST_CONFIG_H += $(QUANTUM_DIR)/rgblight/rgblight_post_config.h
+        COMMON_FEATURES_POST_CONFIG_H += $(QUANTUM_DIR)/rgblight/rgblight_post_config.h
         OPT_DEFS += -DRGBLIGHT_ENABLE
         SRC += $(QUANTUM_DIR)/color.c
         SRC += $(QUANTUM_DIR)/rgblight/rgblight.c
@@ -492,7 +492,7 @@ ifneq ($(strip $(DEBOUNCE_TYPE)), custom)
 endif
 
 ifeq ($(strip $(SPLIT_KEYBOARD)), yes)
-    POST_CONFIG_H += $(QUANTUM_DIR)/split_common/post_config.h
+    COMMON_FEATURES_POST_CONFIG_H += $(QUANTUM_DIR)/split_common/post_config.h
     OPT_DEFS += -DSPLIT_KEYBOARD
     CRC_ENABLE := yes
 
```